uaclient/api/u/pro/attach/auto/full_auto_attach/v1.py

```diff
@@ -0,0 +1,126 @@
+from typing import List, Optional, Tuple
+
+from uaclient import actions, contract, entitlements, messages
+from uaclient.api import exceptions
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.clouds import AutoAttachCloudInstance  # noqa: F401
+from uaclient.config import UAConfig
+from uaclient.data_types import DataObject, Field, StringDataValue, data_list
+from uaclient.entitlements.entitlement_status import CanEnableFailure
+
+
+class FullAutoAttachOptions(DataObject):
+    fields = [
+        Field("enable", data_list(StringDataValue), False),
+        Field("enable_beta", data_list(StringDataValue), False),
+    ]
+
+    def __init__(
+        self,
+        enable: Optional[List[str]] = None,
+        enable_beta: Optional[List[str]] = None,
+    ):
+        self.enable = enable
+        self.enable_beta = enable_beta
+
+
+class FullAutoAttachResult(DataObject, AdditionalInfo):
+    pass
+
+
+def _get_auto_attach_token(cfg: UAConfig, cloud: AutoAttachCloudInstance):
+    contract_client = contract.UAContractClient(cfg)
+    try:
+        tokenResponse = contract_client.request_auto_attach_contract_token(
+            instance=cloud
+        )
+    except exceptions.ContractAPIError as e:
+        if e.code and 400 <= e.code < 500:
+            raise exceptions.NonAutoAttachImageError(
+                messages.UNSUPPORTED_AUTO_ATTACH
+            )
+        raise e
+
+    token = tokenResponse["contractToken"]
+    return token
+
+
+def _is_any_beta(cfg: UAConfig, ents: List[str]) -> Tuple[bool, str]:
+    for name in ents:
+        try:
+            ent_cls = entitlements.entitlement_factory(cfg, name)
+            if ent_cls.is_beta:
+                return True, name
+        except exceptions.EntitlementNotFoundError:
+            continue
+    return False, ""
+
+
+def full_auto_attach(options: FullAutoAttachOptions) -> FullAutoAttachResult:
+    return _full_auto_attach(options, UAConfig(root_mode=True))
+
+
+def _full_auto_attach(options: FullAutoAttachOptions, cfg: UAConfig):
+    instance = actions.get_cloud_instance(cfg)
+    allow_enable = options.enable is None and options.enable_beta is None
+    token = _get_auto_attach_token(cfg, instance)  # type: ignore
+    actions.attach_with_token(cfg, token=token, allow_enable=allow_enable)
+    if allow_enable:
```

Comment:
Can we change the name of this variable to
```suggestion
    if enable_default_services:
```